server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -323,6 +357,11 @@ public XContentBuilder snapshotsToXContent(final XContentBuilder builder) throws
             builder.endObject();
         }
         builder.endObject();
+
+        if (version.onOrAfter(SnapshotsService.SHARD_GEN_IN_REPO_DATA_VERSION)) {
+            shardGenerations.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        }
```

Comment:
`else` shardGenerations should be empty, right? Can we assert that?

Reply:
Right, added that assertion.